lib/nuntiux/mocker.ex

```diff
@@ -326,71 +331,121 @@ defmodule Nuntiux.Mocker do
     :ok
   end
 
+  # credo:disable-for-lines:51 Credo.Check.Refactor.ABCSize
   @doc false
-  @spec maybe_run_expects(message, expects) :: expects_matched
+  @spec maybe_run_expects(message, expects, opts) :: mocked_stack_with
         when message: term(),
              expects: expects(),
-             expects_matched: expects_matched()
-  defp maybe_run_expects(message, expects) do
-    Enum.reduce_while(
-      expects,
-      @label_nomatch,
-      fn
-        {_expect_id, _expect_fun}, {@label_match, _matched} = result ->
-          {:halt, result}
-
-        {_expect_id, expect_fun}, @label_nomatch ->
-          try do
-            {:cont, {@label_match, expect_fun.(message)}}
-          catch
-            :error, :function_clause ->
-              {:cont, @label_nomatch}
-          end
+             opts: opts(),
+             mocked_stack_with: mocked_stack_with()
+  defp maybe_run_expects(message, expects, opts) do
+    no_match_fun = fn _msg -> @label_nomatch end
+    expects = Enum.reverse([{:_, no_match_fun} | Map.to_list(expects)])
+
+    mocked_stack_with =
+      Enum.reduce_while(
+        expects,
+        mocked_stack_with(false),
+        fn
+          {expect_id, expect_fun}, %{mocked?: false, stack: stack_trace} ->
+            try do
+              expect_fun.(message)
+            else
+              @label_nomatch ->
+                {:halt, mocked_stack_with(false, stack_trace)}
+
+              match ->
+                {:halt, mocked_stack_with(true, stack_trace, match)}
```

Comment:
⚠️ we stop on the first match. I see no note on the README regarding this, but I don't think there's a situation where you'd want a message to be accepted by many "expects", atm.

Reply:
Yeah. I agree.